app/Http/Controllers/SignupsController.php

```diff
@@ -124,6 +124,17 @@ public function index(Request $request)
      */
     public function show(Request $request, Signup $signup)
     {
+        // Only allow an admin or the user who owns the signup to see the signup's unapproved posts.
+        if (starts_with($request->query('include'), 'posts')) {
+            $types = (new \League\Fractal\Manager)
+                ->parseIncludes($request->query('include'))
+                ->getIncludeParams('posts');
+
+            $types = $types ? $types->get('type') : null;
+
+            $signup = $signup->withVisiblePosts($types)->first();
```

Comment:
This `->first()` is going to cause us trouble I think - this is equivalent to calling `Signup::first();` so it'll always rewrite `$signup` as the first signup in the database, rather than whatever was passed from the route we're on.

Reply:
It's a little tricky that the test suite is still passing here - let's edit one of the `v3/signup/{id}` tests to seed a few more signups and grab one in the middle so we let our test suite catch an error like this.

Reply:
ohhh I thought that `first()` called the first response back! Thanks for flagging that @DFurnes ! I was testing with the first signup so that actually makes a lot of sense. 

If I just do `$signup->withVisiblePosts($types)` I get this error:
```
  "message": "Type error: Argument 1 passed to Rogue\\Http\\Transformers\\SignupTransformer::transform() must be an instance of Rogue\\Models\\Signup, instance of Illuminate\\Database\\Eloquent\\Builder given, called in /home/vagrant/DoSomething.org/rogue/vendor/league/fractal/src/Scope.php on line 338",
    "exception": "Symfony\\Component\\Debug\\Exception\\FatalThrowableError",
``` 
I'm not sure how to access this scope on the `$signup` without passing a query (instead of just the signup)?

Reply:
oh wait I think I found a fix! :)

Reply:
@DFurnes also added new tests to test this logic as well! Thanks for flagging that! Ready for another 👀 when you have a sec!